pkg/cli/config/config.go

```diff
@@ -58,19 +60,29 @@ func GetBackplaneConfiguration() (bpConfig BackplaneConfiguration, err error) {
 		}
 	}
 
-	// Check if user has explicitly defined backplane URL; it has higher precedence over the config file
-	err = viper.BindEnv("url", info.BackplaneURLEnvName)
+	// Check if user has explicitly defined proxy; it has higher precedence over the config file
+	err = viper.BindEnv("proxy-url", info.BackplaneProxyEnvName)
 	if err != nil {
 		return bpConfig, err
 	}
 
-	// Check if user has explicitly defined proxy; it has higher precedence over the config file
-	err = viper.BindEnv("proxy-url", info.BackplaneProxyEnvName)
-	if err != nil {
+	// Warn user if url defined in the config file
+	if viper.GetString("url") != "" {
+		logger.Warn("URL configuration is deprecated, Pls remove backplane config URL setting")
+	}
+
+	// Fetch backplane URL from ocm env
```

Comment:
Just noticed this, if I understand the flow correctly then the OCM API call is made regardless if the env var is set or not. How about we switch the flow?

i.e. **If** the env var is present then assign the config URL to that; **if else** then make the call to the OCM API endpoint. We'll be saving up on API calls in such case :smile:

Reply:
That's a great point and fixed it ..